cylc/flow/workflow_files.py

```diff
@@ -1311,7 +1311,7 @@ def _parse_src_reg(reg: Path, cur_dir_only: bool = False) -> Tuple[Path, Path]:
     return (reg, abs_path)
 
 
-def validate_workflow_name(name: str) -> None:
+def validate_workflow_name(name: str, runNcheck=False) -> None:
```

Comment:
Why do we need this `runNcheck` argument?

Reply:
Other commands that use `validate_workflow_name` can (sometimes should) be passed a name which ends in `runN` or `runX`: Consider `cylc clean `~/cylc-run/my-workflow/runN`. Cylc clean uses validate workflow, and in that case it certainly doesn't want to claim that `~/cylc-run/my-workflow/runN` is not valid.